src/main/scala/com/databricks/spark/xml/parsers/StaxXmlParser.scala

```diff
@@ -90,6 +90,9 @@ private[xml] object StaxXmlParser {
             failedRecord(xml)
           case _: XMLStreamException =>
             failedRecord(xml)
+          case _: ClassCastException =>
```

Comment:
How about  just catching `RuntimeException` and deduplicate some exceptions here?

Reply:
Changed all exception handling to :

case NonFatal(e) =>
            failedRecord(xml)